crates/gitbutler-branch-actions/src/virtual.rs

```diff
@@ -1020,35 +1022,58 @@ impl<'repo, 'cache, 'graph> IsCommitIntegrated<'repo, 'cache, 'graph> {
             .maybe_find_branch_by_refname(&target.branch.clone().into())?
             .ok_or(anyhow!("failed to get branch"))?;
         let remote_head = remote_branch.get().peel_to_commit()?;
-        let mut upstream_commits =
-            ctx.repository()
-                .l(remote_head.id(), LogUntil::Commit(target.sha), false)?;
-        upstream_commits.sort();
         let upstream_tree_id = ctx.repository().find_commit(remote_head.id())?.tree_id();
+
+        let upstream_commits =
+            ctx.repository()
+                .log(remote_head.id(), LogUntil::Commit(target.sha), true)?;
+        let upstream_change_ids = upstream_commits
+            .iter()
+            .filter_map(|commit| commit.change_id())
+            .sorted()
+            .collect();
+        let upstream_commits = upstream_commits
```

Comment:
For consistency, would this be better named `upstream_commit_ids`, as to not have some `commit` variables be `git2::Oid`, and some `git2::Commit`?